### ISSUE TITLE
New Line Settings for DataDisplay

### ISSUE THREAD

**Issue by KarlZeilhofer** [neundorf/CuteCom#29]
Hi, 
I just talked to a device which sends only CR at the end of a line. 
This leads to an unreadable visual representation. 

I'd like to provide you with a pull request, but I didn't find a proper way to access the MainWindow's m_settings from the DataDisplay. 

I suggest to replace the occurrences of '\n' in ```void DataDisplay::displayData(const QByteArray &data)``` by '\r', if and only if the Settings::m_lineterm is CR. 

Greets, 
Karl

**Comment by cyc1ingsir** [comment]
Hi Karl,
I've just added an option to add an CR at the end of the line at my testing "framework" for the data display
https://github.com/cyc1ingsir/DataDisplayTester/tree/searchPanel
I wouldn't call it unreadable but a CR will not start on a new line. That's right.
![DataDisplay with CR](https://cloud.githubusercontent.com/assets/2297037/21549185/dc5bc97a-cdf0-11e6-9ce8-93a9517d243b.jpg)

Looking at the code 
```c++
// print one newline for \r\n only
        if ((isprint(b)) || (b == '\n') || (b == '\r') || (b == '\t')) {

            if (b == '\r') {
                if (m_displayCtrlCharacters)
                    line.data += QChar(0x240D);
            } else if (b == '\n') {
                if (m_displayCtrlCharacters)
                    line.data += QChar(0x240A);
                line.data += '\n';
```
this was done on purpose.
If coupled with the m_lineterm setting a assumtion is made that the device expects the input terminated with CR if it terminates the output lines with CR only too.
To tackle your issue, DataDisplay needs to be added with a method setting another flag which has be to evaluated within the code snippet shown above.

Although I don't really expect any differences: Would you mind testing CuteCom from this branch:
https://github.com/cyc1ingsir/cutecom/tree/test/search 
and tell me if it makes things looking better?
I've made quite a few changes to DataDisplay (and to be honest would prefer a PR - including changes to the DataDisplay - be made against this branch :-) )
Also I would highly appreciate the rewrite being broadly tested.

Thanks,
Meinhard

**Comment by KarlZeilhofer** [comment]
It still looks the same with the /test/search branch:
![2016-12-30_002](https://cloud.githubusercontent.com/assets/2773445/21560410/384b54a8-ce5f-11e6-92bd-91f242672de4.png)

I’ll try to provide you a PR.

**Comment by KarlZeilhofer** [comment]
I created a pullreqest there:
https://github.com/cyc1ingsir/cutecom/pull/2

please note the bug on misalignment of data and time.
![2016-12-30_003](https://cloud.githubusercontent.com/assets/2773445/21561483/62d466c0-ce6d-11e6-9c67-eaa6d9a03bcd.png)

**Comment by cyc1ingsir** [comment]
Thanks for your effort.
Your changes are included in the PR #31 and will lead to release 0.40.0 in the near future.
Feel free to test.
